examples/chatgpt/gpt_actions_library/gpt_action_mongodb_vector.md

```diff
@@ -0,0 +1,356 @@
+  description: API for searching movie plots and titles based on a search term.
+  version: 1.0.0
+servers:
+  - url: https://us-central1-enhanced-storm-404515.cloudfunctions.net
+    description: Main production server
+paths:
+  /girish-mongo-search:
+    post:
+      operationId: searchDatabase
+      summary: Search for movies based on a search term.
+      description: |
+        This endpoint accepts a search term and returns a list of movies with their plot, title, and a relevance score.
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                searchTerm:
+                  type: string
+                  description: The term to search for in movie plots and titles.
+              required:
+                - searchTerm
+      responses:
+        '200':
+          description: A list of movies matching the search term.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: object
+                  properties:
+                    plot:
+                      type: string
+                      description: The plot of the movie.
+                    title:
+                      type: string
+                      description: The title of the movie.
+                    score:
+                      type: number
+                      format: float
+                      description: The relevance score of the movie.
+        '400':
+          description: Bad request. Occurs if the request body is malformed or missing required fields.
+        '500':
+          description: Internal server error. Occurs if there is a server-side issue.
+
+```
+
+## Authentication Instructions
+
+Below are instructions on setting up authentication with this 3rd party application. Have questions? Check out [Getting Started Example](https://platform.openai.com/docs/actions/getting-started) to see how this step works in more detail.
+
+
+### In Google Cloud Console
+In Google Cloud Console, you need to create OAuth client ID credentials. To navigate to the right page search for "Credentials" in Google Cloud Console or enter `https://console.cloud.google.com/apis/credentials?project=<your_project_id>` in your browser. You can read more about it [here](https://developers.google.com/workspace/guides/create-credentials).
+
+Click on "CREATE CREDENTIALS" and select "Oauth client ID". Select "Web Application" for "Application type" and enter the name of your application (see below).
+
+![](../../../images/gcp-function-middleware-oauthclient.png)
+
+In the "OAuth client created" modal dialog, please take note of the
+
+* Client ID
+* Client secret
+
+
+### In ChatGPT (refer to Step 2 in the Getting Started Example)
+
+In ChatGPT, click on "Authentication" and choose **"OAuth"**. Enter in the information below.
```

Comment:
Wouldn't we want to leverage a MongoDB Oauth connection, if possible? https://www.mongodb.com/docs/atlas/security-oidc/

Does it enable all authenticated users to Google (regardless of their directory) to run the function?